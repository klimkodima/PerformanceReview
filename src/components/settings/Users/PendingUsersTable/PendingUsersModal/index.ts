export { default as PendingUsersModal } from './PendingUsersModal';
