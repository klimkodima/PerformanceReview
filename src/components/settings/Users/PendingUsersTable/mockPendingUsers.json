[
  {
    "id": 5,
    "fullName": "Pety Petrov",
    "username": "Pety Petrov",
    "email": "p.petrov@gpsolutions.com",
    "enabled": true,
    "roleName": "",
    "pending": true,
    "jiraAccountId": "12866hg",
    "worksFrom": "08-31-2015",
    "thirdParty": false
  },
  {
    "id": 6,
    "fullName": "Egor Egorov",
    "username": "Egor Egorov",
    "email": "e.egorov@gpsolutions.com",
    "enabled": false,
    "roleName": "",
    "teamId": 1,
    "teamName": "X-Rays",
    "level": "TEAM_LEAD",
    "pending": true,
    "jiraAccountId": "12866hg",
    "worksFrom": "08-31-2015",
    "thirdParty": false
  }
]
