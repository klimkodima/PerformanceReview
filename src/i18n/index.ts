import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import common from './locales.en.json';

const resources = {
  en: {
    translation: common
  }
};

const setI18nextConfig = async () =>
  await i18n.use(LanguageDetector).use(initReactI18next).init({
    resources,
    fallbackLng: 'en'
  });

void setI18nextConfig();

export default i18n;
