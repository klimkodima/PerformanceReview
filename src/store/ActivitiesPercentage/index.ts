export {
  default as activitiesPercentageReducer,
  fetchAuditActivities,
  setAuditActivities,
  setTeamActivities,
  fetchTeamActivities
} from './reducer';
export { activitiesPercentageSaga } from './sagas';
export { selectAuditActivities, selectTeamActivities } from './selectors';

export type { AuditActivitiesType, ActivityLabelType } from './types';
