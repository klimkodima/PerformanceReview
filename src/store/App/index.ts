export { selectAvailableWidgets, selectSettingsPermission } from './selectors';
export {
  default as appReducer,
  setAvailableWidgets,
  fetchAvailableWidgets
} from './reducer';
export { appSaga } from './sagas';
export type { AvailableWidgetsType, SettingsPermissionType } from './types';
