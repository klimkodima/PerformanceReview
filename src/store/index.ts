import createSagaMiddleware from 'redux-saga';
import { configureStore } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import rootReducer from './rootReducer';
import rootSaga from './rootSaga';

const sagaMiddleware = createSagaMiddleware();

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['dateFilter', 'tasksCoefficient']
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER]
      }
    }).concat(sagaMiddleware)
});

sagaMiddleware.run(rootSaga);

export type RootStateType = ReturnType<typeof rootReducer>;
export type StoreType = typeof store;

export const persistor = persistStore(store);
export default store;
