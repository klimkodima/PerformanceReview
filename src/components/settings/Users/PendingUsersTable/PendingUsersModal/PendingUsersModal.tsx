import { ReactElement, memo } from 'react';

import {
  Checkbox,
  FormControl,
  InputLabel,
  Select,
  Table,
  TableBody,
  TableCell,
  TableRow
} from '@mui/material';
import AccountCircleIcon from '@mui/icons-material/AccountCircle';
import InfoIcon from '@mui/icons-material/Info';

import { createFormattedString } from 'src/utils';
import { InfoTooltip, Modal } from 'src/components/common';
import {
  ACTIVATE_BUTTON,
  ADMIN_ROLE,
  AUDITOR_ROLE,
  CANCEL_BUTTON,
  MANAGER_ROLE,
  MODAL_INFO_TOOLTIP_TEXT,
  MODAL_PRESET_TITLE,
  MODAL_SUBTITLE,
  MODAL_TITLE,
  OPTION_1,
  OPTION_2,
  OPTION_3,
  OPTION_4,
  PRESET_WORKING_HOURS_KEY,
  SELECT_LABEL,
  TEAM_LEAD_LEVEL,
  TEAM_LEAD_ROLE
} from '../../constants';
import { onChangePendingUsers } from '../../helpers';

import { UserInfoType } from 'src/store/Users/types';
import { UpdatedPendingUsersType } from '../../types';

type PendingUsersModalPropsType = {
  selectedUsers: UserInfoType[];
  usersSelectedInfo: UpdatedPendingUsersType[];
  setUsersSelectedInfo: (value: UpdatedPendingUsersType[]) => void;
  onCloseClick: () => void;
  onSaveClick: () => void;
  isDisabledSendBtn: boolean;
};

const PendingUsersModal = memo(
  ({
    selectedUsers,
    usersSelectedInfo,
    setUsersSelectedInfo,
    onCloseClick,
    onSaveClick,
    isDisabledSendBtn
  }: PendingUsersModalPropsType): ReactElement => (
    <Modal
      isShowFooterButtons={true}
      className='pending-users-modal'
      saveButtonText={ACTIVATE_BUTTON}
      closeButtonText={CANCEL_BUTTON}
      onCloseClick={onCloseClick}
      onSaveClick={onSaveClick}
      isDisabledSaveButton={isDisabledSendBtn}
    >
      <h2>{MODAL_TITLE}</h2>
      <p>{MODAL_SUBTITLE}</p>
      <div className='pending-users-modal__table-container'>
        <Table aria-label='table'>
          <TableBody>
            {selectedUsers.map(({ id, fullName, email, level, teamName }) => (
              <TableRow key={id}>
                <TableCell>
                  <AccountCircleIcon />
                  {fullName}
                </TableCell>
                <TableCell>{email}</TableCell>
                <TableCell>
                  {level ? createFormattedString(level) : '-'}
                </TableCell>
                <TableCell>
                  {teamName ? createFormattedString(teamName) : '-'}
                </TableCell>
                <TableCell className='pending-users-modal__icon-block'>
                  <FormControl fullWidth>
                    {(!level || !teamName) && (
                      <InputLabel id='pending-users-modal-label'>
                        {SELECT_LABEL}
                      </InputLabel>
                    )}
                    <Select
                      data-testid='pending-users-modal-select'
                      native
                      defaultValue={
                        level === TEAM_LEAD_LEVEL
                          ? TEAM_LEAD_ROLE
                          : AUDITOR_ROLE
                      }
                      id='pending-users-modal-label'
                      name='role'
                      label={!level || !teamName ? SELECT_LABEL : ''}
                      onChange={(e) =>
                        onChangePendingUsers(
                          e,
                          id,
                          usersSelectedInfo,
                          setUsersSelectedInfo
                        )
                      }
                    >
                      {(!level || !teamName) && (
                        <option aria-label='None' value='' />
                      )}
                      {(level || teamName) && (
                        <>
                          <option value={AUDITOR_ROLE}>{OPTION_1}</option>
                          <option value={TEAM_LEAD_ROLE}>{OPTION_2}</option>
                        </>
                      )}
                      <option value={MANAGER_ROLE}>{OPTION_3}</option>
                      <option value={ADMIN_ROLE}>{OPTION_4}</option>
                    </Select>
                  </FormControl>
                  {(!level || !teamName) && (
                    <InfoTooltip
                      title={MODAL_INFO_TOOLTIP_TEXT}
                      icon={
                        <InfoIcon
                          className='criteria-item__icon'
                          color='primary'
                          data-testid='criteria-item-info-icon'
                        />
                      }
                    />
                  )}
                </TableCell>
                <TableCell>
                  <Checkbox
                    data-testid='pending-users-modal-checkbox'
                    inputProps={{ 'aria-label': 'checkbox' }}
                    color='success'
                    name={PRESET_WORKING_HOURS_KEY}
                    onChange={(e) =>
                      onChangePendingUsers(
                        e,
                        id,
                        usersSelectedInfo,
                        setUsersSelectedInfo
                      )
                    }
                  />
                  {MODAL_PRESET_TITLE}
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </div>
    </Modal>
  )
);

export default PendingUsersModal;

PendingUsersModal.displayName = 'PendingUsersModal';
