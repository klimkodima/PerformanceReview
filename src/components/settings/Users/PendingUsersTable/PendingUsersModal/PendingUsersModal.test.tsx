import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';

import { Provider } from 'react-redux';

import store from 'src/store';
import i18n from 'src/i18n';
import PendingUsersModal from './PendingUsersModal';
import mockPendingUsers from '../mockPendingUsers.json';

const mockSelectedPendingUsersInfo = [
  {
    id: 5,
    thirdParty: false,
    role: ''
  },
  {
    id: 6,
    thirdParty: false,
    role: ''
  }
];

describe('PendingUsersModal tests', () => {
  test('PendingUsersModal snapshot', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'portal');
    document.body.appendChild(root);

    const setUsersSelectedInfo = jest.fn();
    const onCloseClick = jest.fn();
    const onSaveClick = jest.fn();

    const { getByTestId } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <PendingUsersModal
            setUsersSelectedInfo={setUsersSelectedInfo}
            selectedUsers={mockPendingUsers}
            usersSelectedInfo={mockSelectedPendingUsersInfo}
            onCloseClick={onCloseClick}
            onSaveClick={onSaveClick}
            isDisabledSendBtn={true}
          />
        </I18nextProvider>
      </Provider>
    );

    expect(getByTestId('modal-overlay')).toMatchSnapshot();
  });
});
