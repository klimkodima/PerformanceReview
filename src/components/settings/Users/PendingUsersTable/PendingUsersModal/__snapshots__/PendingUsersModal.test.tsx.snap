// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingUsersModal tests PendingUsersModal snapshot 1`] = `
<div
  class="modal-overlay"
  data-testid="modal-overlay"
>
  <div
    class="modal-body pending-users-modal"
  >
    <h2>
      Specify the users' settings
    </h2>
    <p>
      Please, select a role for each user. Some roles can be preselected based on the user's level - you can change the selection, if necessary. Also, you can mark the user as "Preset working hours" if this user does not fill in the corporate Jira. In this case, the system will use the time set for such users in the special settings table "Preset working hours".
    </p>
    <div
      class="pending-users-modal__table-container"
    >
      <table
        aria-label="table"
        class="MuiTable-root css-rqglhn-MuiTable-root"
      >
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AccountCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
              Pety Petrov
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              p.petrov@gpsolutions.com
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              -
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              -
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium pending-users-modal__icon-block css-1ex1afd-MuiTableCell-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  id="pending-users-modal-label"
                >
                  Select a role
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  data-testid="pending-users-modal-select"
                >
                  <select
                    aria-invalid="false"
                    class="MuiNativeSelect-select MuiNativeSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-ffg8md-MuiNativeSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    id="pending-users-modal-label"
                    name="role"
                  >
                    <option
                      aria-label="None"
                      value=""
                    />
                    <option
                      value="MANAGER"
                    >
                      Manager
                    </option>
                    <option
                      value="ADMIN"
                    >
                      Admin
                    </option>
                  </select>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Select a role
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <svg
                aria-hidden="true"
                aria-label=" “This user misses team/level. You can select only Manager or Admin roles for such users“"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium criteria-item__icon css-ci5apu-MuiSvgIcon-root"
                data-mui-internal-clone-element="true"
                data-testid="criteria-item-info-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                />
              </svg>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <span
                class="MuiCheckbox-root MuiCheckbox-colorSuccess MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSuccess PrivateSwitchBase-root css-10drtbx-MuiButtonBase-root-MuiCheckbox-root"
                data-testid="pending-users-modal-checkbox"
              >
                <input
                  aria-label="checkbox"
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  name="thirdParty"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              Preset working hours
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AccountCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
              Egor Egorov
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              e.egorov@gpsolutions.com
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              Team lead
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              X-rays
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium pending-users-modal__icon-block css-1ex1afd-MuiTableCell-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  data-testid="pending-users-modal-select"
                >
                  <select
                    aria-invalid="false"
                    class="MuiNativeSelect-select MuiNativeSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-ffg8md-MuiNativeSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    id="pending-users-modal-label"
                    name="role"
                  >
                    <option
                      value="AUDITOR"
                    >
                      Auditor
                    </option>
                    <option
                      selected=""
                      value="TEAM_LEADER"
                    >
                      Team leader
                    </option>
                    <option
                      value="MANAGER"
                    >
                      Manager
                    </option>
                    <option
                      value="ADMIN"
                    >
                      Admin
                    </option>
                  </select>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-hdw1oc"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              <span
                class="MuiCheckbox-root MuiCheckbox-colorSuccess MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSuccess PrivateSwitchBase-root css-10drtbx-MuiButtonBase-root-MuiCheckbox-root"
                data-testid="pending-users-modal-checkbox"
              >
                <input
                  aria-label="checkbox"
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  name="thirdParty"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              Preset working hours
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="modal-overlay__buttons"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-sxix9q-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            />
          </svg>
        </span>
        Activate Users
      </button>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-180lf2n-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
        Cancel
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
